fix: return 404 if built_in_database not configured as auth source (#11797)
emqx/emqx: changes/ce/fix-10577.en.md

Reviewer: Is the PR number of this file correct?

Author: Oh, yeah, no … it’s the ticket number 🤦

> HTTP API to return status code `404` instead of `20X` if 'built_in_database' not configured as authorization source.

Reviewer: ```suggestion
HTTP API now returns status code `404` instead of `20X` if `built_in_database` is not configured as authorization source.
```

> HTTP API to return status code `404` instead of `20X` if 'built_in_database' not configured as authorization source.

Reviewer 2: nit:
Modified HTTP API behavior for APIs managing the `built_in_database` authorization source: They will now return a `404` status code if 'built_in_database' is not set as the authorization source, replacing the former `20X` response.
